### PR TITLE
Add Simulation visualization and Fix turret locking

### PR DESCRIPTION
**Pull request by alicekuznetsov**
AdvantageScope_(WPILib)_T4vyzHQluU

I exported the hood wrong so that's wrong in this clip... I need to fix that soon and then we'll talk about where to put the model asset. I'll probably just throw it in assets like last year. My drivebase is also missing colsons. I don't know whats up with that. AAND i need an icat. also don't know know whats up with that. 

All of these todos aren't major, especially compared to the Turret Locking fix. If we want that in main, I'd be fine with merging this in and making these a separate issue

TODO:
- [ ] Re-get hood model (the origin NEEDS to be in the center of the pivot point, and the planes need to be correct)
- [ ] Test hood positioning in sim
- [ ] Maybe fix the drivetrain model so i actually have wheels?
- [ ] Add ICAT models
- [ ] Upload robot model somewhere

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Turret.java [hunk 1]
```diff
@@ -263,6 +272,10 @@ public Optional<Rotation2d> getDesiredAngleToLock(Pose2d robotPose, Pose3d[] fie
     // Get the angle of 0,0 to the turret pose
     desiredLockingAngle = new Rotation2d(relativeToTarget.getX(), relativeToTarget.getY());
 
+    // Account for robot rotation
+    desiredLockingAngle = desiredLockingAngle
+        .rotateBy(robotPose.getRotation().minus(new Rotation2d().fromDegrees(180)));
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Turret.java]: This "minus" is kinda weird and i really don't like it but it works. If anyone else can think or congure up a better way of doing this (I'm sure that there _is_ a better way, i just cant think of it right now), i am taking suggestions

**Comment by TaylerUva** [comment]
Where are the 3D models stored?

**Comment by alicekuznetsov** [comment]
Currently, just on my computer (but the hood is still wrong)

Do we want them in the assets folder or somewhere else?

**Comment by TaylerUva** [comment]
> Do we want them in the assets folder or somewhere else?

Depends on the size. Maybe as git lfs if they are large

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**